src/rayTracing.py

```diff
@@ -0,0 +1,127 @@
+#  * https://github.com/tavianator/dimension                               *
+#  * Copyright (C) 2020 Pero Prebeg <Pero.Prebeg@fsb.hr>                   *
+#  *                                                                       *
+#  * The content of this file is part of The Dimension Library. The        *
+#  * original C Code of The Dimensions Library has been translated to      *
+#  * Python code, further accomodations have been made to emphasize the    *
+#  * purpose of d3v.                                                       *
+#  *                                                                       *
+#  * The Dimension Library is free software; you can redistribute it and/  *
+#  * or modify it under the terms of the GNU Lesser General Public License *
+#  * as published by the Free Software Foundation; either version 3 of the *
+#  * License, or (at your option) any later version.                       *
+#  *                                                                       *
+#  * The Dimension Library is distributed in the hope that it will be      *
+#  * useful, but WITHOUT ANY WARRANTY; without even the implied warranty   *
+#  * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU  *
+#  * Lesser General Public License for more details.                       *
+#  *                                                                       *
+#  * You should have received a copy of the GNU Lesser General Public      *
+#  * License along with this program.  If not, see                         *
+#  * <http://www.gnu.org/licenses/>.                                       *
+#  *************************************************************************/
+
+import math
+
+from bounds import BBox
+from pyVector3 import pyvector3
+
+
+class dmnsn_ray:
+    def __init__(self, los):
+        self._x0 = pyvector3()
+        self._x0.setFromQt(los[0])  # P0
+        self._n = pyvector3()
+        self._n.setFromQt(los[1])  # K
+
+    @property
+    def x0(self):
+        return self._x0
+
+    @property
+    def n(self):
+        return self._n
+
+
+class dmnsn_optimized_ray:
+    def __init__(self, ray: dmnsn_ray):
+        self._x0 = ray.x0
+        self._n_inv = pyvector3()
+        self._n_inv.setFromScalars((1.0 / ray.n.X) if ray.n.X != 0 else math.inf,
+                                   (1.0 / ray.n.Y) if ray.n.X != 0 else math.inf,
+                                   (1.0 / ray.n.Z) if ray.n.Z != 0 else math.inf)
+
+    @property
+    def x0(self):
+        return self._x0
+
+    @property
+    def n_inv(self):
+        return self._n_inv
+
+
+class dmnsn_aabb:
+    def __init__(self):
+        self._min = pyvector3()
+        self._max = pyvector3()
+
+    def setFromBBox(self, box: BBox):
+        self._min.setFromNp(box.minCoord)
+        self._max.setFromNp(box.maxCoord)
+
+    def dmnsn_ray_box_intersection(self, optray: dmnsn_optimized_ray, t):
```

Comment:
This lacks explanation of the algorithm.....
Also, what is the purpose of parameter `t`? It is obviously controlled from the outside, does it have any connection to `optarray.x0`? 

Why is the `dmnsn_optimized_ray` used as  parameter (instead of constructing it inside the function)? Does it  mean that one have to manually construct object of that type just to check box-ray intersection?

Reply:
Done by adding some comments and a link of the reference of the algorithm. Parameteter is not necessary and removed.